api/agent_test.go

```diff
@@ -110,7 +110,31 @@ func TestAgent_ForceLeave(t *testing.T) {
 	err := a.ForceLeave("nope")
 	must.NoError(t, err)
 
-	// TODO: test force-leave on an existing node
+	// Force-leave on an existing node
+	_, s2 := makeClient(t, nil, func(c *testutil.TestServerConfig) {
+		c.Server.BootstrapExpect = 0
+	})
+	defer s2.Stop()
+	// Create a new node to join
+	n, err := a.Join(s2.SerfAddr)
+	must.NoError(t, err)
+	must.One(t, n)
+
+	membersBefore, err := a.MembersOpts(&QueryOptions{})
+	must.Eq(t, membersBefore.Members[1].Status, "alive")
+
+	err = a.ForceLeave(membersBefore.Members[1].Name)
+	must.NoError(t, err)
+
+	time.Sleep(3 * time.Second)
+
+	membersAfter, err := a.MembersOpts(&QueryOptions{})
+
+	for _, node := range membersAfter.Members {
+		if node.Name == membersBefore.Members[1].Name {
+			must.Eq(t, node.Status, "leaving")
+		}
+	}
```

Comment:
For this kind of thing, rather than waiting a static period of time, we can use [`shoenig/test/wait`](https://pkg.go.dev/github.com/shoenig/test@v0.6.1/wait) to wait until the assertion is true or the timeout expires. That lets us move the test along if the desired state is reached quickly.

For example, the following takes <3sec on my machine instead of >6sec. This adds up given the number of tests we run!

```suggestion
	f := func() error {
		membersAfter, err := a.MembersOpts(&QueryOptions{})
		if err != nil {
			return err
		}

		for _, node := range membersAfter.Members {
			if node.Name == membersBefore.Members[1].Name {
				if node.Status != "leaving" {
					return fmt.Errorf("node did not leave")
				}
			}
		}
		return nil
	}

	must.Wait(t, wait.InitialSuccess(
		wait.ErrorFunc(f),
		wait.Timeout(3*time.Second),
		wait.Gap(100*time.Millisecond),
	))
```

Reply:
Thanks alot for the suggestion and the code snippet. This is exactly what I'm looking for when writing this test. Totally agree that we can greatly reduce the test run time by making the `wait.Gap()` more aggressive instead of waiting for the whole duration.